src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -0,0 +1,151 @@
+package teammates.client.scripts.sql;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.googlecode.objectify.cmd.Query;
+
+import teammates.storage.entity.Course;
+import teammates.storage.entity.CourseStudent;
+import teammates.storage.sqlentity.Section;
+import teammates.storage.sqlentity.Student;
+import teammates.storage.sqlentity.Team;
+
+/**
+ * Data migration class for course entity.
+ */
+@SuppressWarnings("PMD")
+public class DataMigrationForCourseEntitySql extends
+        DataMigrationEntitiesBaseScriptSql<teammates.storage.entity.Course, teammates.storage.sqlentity.BaseEntity> {
+
+    public static void main(String[] args) {
+        new DataMigrationForCourseEntitySql().doOperationRemotely();
+    }
+
+    @Override
+    protected Query<Course> getFilterQuery() {
+        return ofy().load().type(teammates.storage.entity.Course.class);
+    }
+
+    @Override
+    protected boolean isPreview() {
+        return false;
+    }
+
+    /*
+     * Sets the migration criteria used in isMigrationNeeded.
+     */
+    @Override
+    protected void setMigrationCriteria() {
+        // No migration criteria currently needed.
+    }
+
+    @Override
+    protected boolean isMigrationNeeded(Course entity) {
+        return true;
+    }
+
+    @Override
+    protected void migrateEntity(Course oldCourse) throws Exception {
+        teammates.storage.sqlentity.Course newCourse = createCourse(oldCourse);
+        // TODO: add shutdown hook to save the entity
+        // Runnable shutdownScript = () -> { cascadeDelete(newCourse)};
+        // Runtime.getRuntime().addShutdownHook(new Thread(shutdownScript));
+
+        migrateCourseEntity(newCourse);
+        // verifyCourseEntity(newCourse);
```

Comment:
Please add this as a todo in your PR description so that we can keep track what is left

Reply:
Done!